### PR TITLE
Wt 13 implement login business logic

### PR DESCRIPTION
**Pull request by k1mka**

### CODE REVIEW

File: lib/presentation/screens/sign_up_screen/bloc/sign_up_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:water_tracker/data/repository/repository.dart';
+import 'package:water_tracker/presentation/screens/sign_up_screen/bloc/sign_up_event.dart';
+import 'package:water_tracker/presentation/screens/sign_up_screen/bloc/sign_up_state.dart';
+
+class SignUpBloc extends Bloc<SignUpEvent, SignUpState> {
+  Repository repository;
```
Comment (zddarova) [inline lib/presentation/screens/sign_up_screen/bloc/sign_up_bloc.dart:7]: make final

File: lib/presentation/screens/sing_in_screen/bloc/sign_in_bloc.dart [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:water_tracker/data/repository/repository.dart';
+import 'package:water_tracker/presentation/screens/sing_in_screen/bloc/sign_in_event.dart';
+import 'package:water_tracker/presentation/screens/sing_in_screen/bloc/sign_in_state.dart';
+
+class SignInBloc extends Bloc<SignInEvent, SignInState> {
+  Repository repository;
```
Comment (zddarova) [inline lib/presentation/screens/sing_in_screen/bloc/sign_in_bloc.dart:7]: make final

File: lib/presentation/screens/sing_in_screen/bloc/sign_in_bloc.dart [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+  SignInBloc(this.repository) : super(InitialSignInState()) {
+    on<SignInUserEvent>((event, emit) async {
+      try {
+        emit(LoadingSignInState());
+        await repository.loginUser(event.email, event.password);
+        emit(SuccessfullySingInState());
```
Comment (zddarova) [inline lib/presentation/screens/sing_in_screen/bloc/sign_in_bloc.dart:14]: and if not successful?

File: lib/presentation/widgets/sign_in_form.dart [hunk 4]
```diff
@@ -47,8 +49,10 @@ class _MyLogFormWidgetState extends State<LogoForm> {
           spacer,
           CustomButton(
             onPressed: () {
-              // TODO: call something on validation
-              if (formKey.currentState!.validate()) {}
+              if (formKey.currentState!.validate()) {
+                // TODO: call something on validation
+                //  widget.onSignInButtonPressed();
```
Comment (zddarova) [inline lib/presentation/widgets/sign_in_form.dart:54]: undo comment

**Review by zddarova — APPROVED** [review]

File: lib/presentation/screens/sign_up_screen/bloc/sign_up_bloc.dart (hunk 1 above)
Comment (k1mka) [reply lib/presentation/screens/sign_up_screen/bloc/sign_up_bloc.dart:7]: ok

File: lib/presentation/screens/sing_in_screen/bloc/sign_in_bloc.dart (hunk 2 above)
Comment (k1mka) [reply lib/presentation/screens/sing_in_screen/bloc/sign_in_bloc.dart:7]: ok

File: lib/presentation/screens/sing_in_screen/bloc/sign_in_bloc.dart (hunk 3 above)
Comment (k1mka) [reply lib/presentation/screens/sing_in_screen/bloc/sign_in_bloc.dart:14]: ok

**Review by zddarova — APPROVED** [review]

**Pull request merged**